src/arraytypes/dictencoding.jl

```diff
@@ -113,12 +113,14 @@ function arrowvector(::DictEncodedType, x, i, nl, fi, de, ded, meta; dictencode:
     validity = ValidityBitmap(x)
     if !haskey(de, id)
         # dict encoding doesn't exist yet, so create for 1st time
+        ET = encodingtype(length(DataAPI.levels(x)))
```

Comment:
I think we want to avoid calling `DataAPI.levels` here because we'll end up doing the "pooling" operation twice: once here, and again below when we call `PooledArray(x, ET)`. 

I think this would be my recommended approach:
```julia
# dict encoding doesn't exist yet, so create for 1st time
        if DataAPI.refarray(x) === x
            # need to encode ourselves
            x = PooledArray(x, encodingtype(length(x)))
            refa = DataAPI.refarray(x)
        else
            refa = copy(DataAPI.refarray(x))
        end
        inds = converter(encodingtype(length(refa)), refa)
```

Reply:
But the minimum size of the index type depends on the number of levels not the length of the vector.  For me a typical case for a DictEncoding can have a small number of levels, such as would be expressible by an Int8, but the size of the vector itself would require an Int32.  Demographic variables like Sex, Race, IncomeLevel, ... are typical cases for me.

May I suggest an alternative which is to add a named Bool argument `signed` to the `PooledArray` constructor which will compress the index size to signed integer types.  I made the suggestion assuming that PooledArrays implemented the `DataAPI` methods like `levels` so it would be easy to determine the number of levels in the call to `converter`.  But now I see it doesn't.  Is there a reason why not?

Reply:
Ah, I see now that it is `DataAPI.refpool` that is implemented in PooledArrays.

So my suggestion is either to change `PooledArrays` to use signed integers for the refarray by default (I think I may have been guilty of suggesting unsigned integers initially in case the size of the refpool was between 128 and 255) or allow for a `signed` named argument in the PooledArray constructor .

In either case, can we use `converter(encodingtype(length(DataAPI.refpool)), refa)` for the inds?

Reply:
Yes, `converter(encodingtype(length(DataAPI.refpool)), refa)` sounds great